### DISCUSSION TITLE
get-chocolateywebfile user/pass

### DISCUSSION THREAD

**Discussion by t794104** [chocolatey/choco#2477] [Q&A] [answered]
It is possible too download file with https://docs.chocolatey.org/en-us/create/functions/get-chocolateywebfile which is behind a authorizition ?

Our interal JFrog Artifactory only allowes access with user/pass of API key. And i would love too download installers by choco and not imbed them in choco packages.

Is this possible

**Comment by TheCakeIsNaOH** [comment, marked as answer]
I am assuming that Artifactory supports HTTP(s) basic authentication for downloading files?

If so, you can specify an `Authorization` header with the required credentials, and pass that to `Get-ChocolateyWebFile` via the `Options` parameter.

Here is what I have used to download files from a Nexus server. You will need to get the `$user` and `$password` into the script. Hardcoding or passing via package parameters should work.
```
$CredsBase64           = [System.Convert]::ToBase64String([System.Text.Encoding]::ASCII.GetBytes("$user:$password"))
$HeaderCreds           = @{ Headers = @{ Authorization = "Basic $CredsBase64" } }

$packageArgs = @{
  packageName   = $env:ChocolateyPackageName
  FileFullPath  = Join-Path $toolsDir 'filename'
  Url           = 'url'
  Checksum      = 'checksum'
  ChecksumType  = 'sha256'
  Options       = $HeaderCreds
}

Get-ChocolateyWebFile @packageArgs
```

**Reply by t794104** [reply]
This should work for me. Thanks
